### PR TITLE
Left panel contextual menu: improve menu items labels and tooltips

### LINKED ISSUE #10670: Context menu say "checkout branch..." but should be "checkout tag ..."
### Environment

- Git Extensions 4.0.2.16100
- Build 25100ec1f7c8da8f798613da74826038af81a50e
- Git 2.38.1.windows.1
- Microsoft Windows NT 10.0.19044.0
- .NET 6.0.13
- DPI 144dpi (150% scaling)
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 7.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

### Issue description

Using the context many on tags is a little bit confusing. The menu item say "Checkout branch...", the tooltip say "checkout this tag" and the window say "checkout revision".

### Steps to reproduce

Select any tag in tags tree.

### Did this work in previous version of GitExtensions?

_No response_

### Diagnostics

_No response_

### PR DESCRIPTION
**Pull request by pmiossec**
* Different menu entries depending on the type handled (local, remote and tag)
* Other tooltip fixes around "Fetch & ..." menu items

Fixes #10670

### For Tag:
* Before

image

* After

image

### For branch
* Before
image

* After

image

### For Remote

* After: 

image

## Test methodology 

- Manual

## Test environment(s) 
- Git Extensions 33.33.33
- Build d97d6df550ad7a983afbffdb40eb589cf16b1d9c (Dirty)
- Git 2.38.1.windows.1
- Microsoft Windows NT 10.0.19045.0
- .NET 6.0.13
- DPI 96dpi (no scaling)
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 7.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by pmiossec** [comment]
@Peter-Svahn A point of view on these changes?

**Review by mstv — APPROVED** [review]
👍

**Comment by pmiossec** [comment]
I hesitate to replace 'Merge...' by 'Merge into current branch...' that I find a lot clearer. What do you think?

**Comment by mstv** [comment]
> I hesitate to replace 'Merge...' by 'Merge into current branch...' that I find a lot clearer. What do you think?

This will make it consistent with the RevisionGrid.

image

**Comment by pmiossec** [comment]
@Peter-Svahn A point of view on these changes?

> This will make it consistent with the RevisionGrid.

Done.

**Comment by pmiossec** [comment]
I just saw that tooltip use "rebase **to**" (like in "Rebase current branch to this tag") whereas I prefer "rebase **on**".
What is the better for you?

**Comment by mstv** [comment]
yes, "rebase on"

**Comment by pmiossec** [comment]
FYI, as there are no rocket science changes, I plan to merge this PR soon...

**Comment by RussKie** [comment]
I'm not going to block, but this makes it inconsistent with the revision
grid menus as @mstv mentioned.

Any reason not to use the tag specific wording?

**Comment by mstv** [comment]
> but this makes it inconsistent with the revision grid menus as @mstv mentioned.

It *is* in sync with the RevisionGrid now IMO.

> Any reason not to use the tag specific wording?

These menu items are shared for all kinds of refs, i.e. local / remote branches and tags.

**Comment by pmiossec** [comment]
>Any reason not to use the tag specific wording?

I pushed 2 new commits depending on what we want to achieve. The results:

* for branch:

image

* for remote branch:

image

* for tag:

image

@RussKie @mstv Point of view?

**Comment by gerhardol** [comment]
> > Any reason not to use the tag specific wording?
> 
> I pushed 2 new commits depending on what we want to achieve. The results:

Delete remote branch twice, separate icons

I want to have checkpout branch for remote

Otherwise fine - not checked for aligning to the grid

**Comment by pmiossec** [comment]
> > > Any reason not to use the tag specific wording?
> > 
> > 
> > I pushed 2 new commits depending on what we want to achieve. The results:
> 
> Delete remote branch twice, separate icons
> 
> I want to have checkpout branch for remote
> 
> Otherwise fine - not checked for aligning to the grid

That was a regression introduced during refactoring 😢 It is as you want...

Fixed and screenshot updated.

File: GitUI/BranchTreePanel/RepoObjectsTree.Designer.cs [hunk 1]
```diff
@@ -388,7 +388,7 @@ private void InitializeComponent()
             this.mnubtnFetchRebase.Name = "mnubtnFetchRebase";
             this.mnubtnFetchRebase.Size = new System.Drawing.Size(266, 26);
             this.mnubtnFetchRebase.Text = "Fetch && Re&base";
-            this.mnubtnFetchRebase.ToolTipText = "Fetch & Rebase current branch to this branch";
+            this.mnubtnFetchRebase.ToolTipText = "Fetch & Rebase current branch on this branch";
```
Comment (mstv) [inline GitUI/BranchTreePanel/RepoObjectsTree.Designer.cs:391]:
```suggestion
            this.mnubtnFetchRebase.ToolTipText = "Fetch then rebase current branch on this branch";
```

**Review by mstv — APPROVED** [review]
The labels are fool-proof now. Have not run.

File: GitUI/BranchTreePanel/RepoObjectsTree.Designer.cs (hunk 1 above)
Comment (pmiossec) [reply GitUI/BranchTreePanel/RepoObjectsTree.Designer.cs]: Done with some other slight improvements on tooltips in this 'RepoObjectsTree.Designer.cs' file.

**Comment by pmiossec** [comment]
I have squashed and I think we are in good state to merge.
I will do it soon if I receive no more comments 😉

**Review by gerhardol — APPROVED** [review]
> The labels are fool-proof now. Have not run.

mstv, nothing is foolproof...

**Pull request merged**